src/v/cluster/archival_metadata_stm.cc

```diff
@@ -897,6 +914,13 @@ ss::future<std::error_code> archival_metadata_stm::do_add_segments(
         b.add_raw_kv(std::move(key_buf), std::move(val_buf));
     }
 
+    if (highest_pid != model::producer_id{}) {
+        iobuf key_buf = serde::to_iobuf(
+          archival_metadata_stm::update_highest_producer_id_cmd::key);
+        iobuf val_buf = serde::to_iobuf(highest_pid());
```

Comment:
I'm not against it, it's a good practice. But I'm also not sure how this would be extended (and if push comes to shove, at that time, we can always deprecate this and add a new one)

Reply:
Ill leave it up to you I don't have a strong opinion about it.